assignments/01-ownership-borrowing/src/basics.rs

```diff
@@ -0,0 +1,88 @@
+/// Take ownership of the passed in string and print it
+fn takes_ownership_and_prints(unowned: String) {
+    println!("{}", unowned);
+}
+
+/// Take a string slice and return the first letter, or None in the case
+/// of a blank string
+fn first_letter(string: &str) -> Option<&str> {
+    if let Some(first_char) = string.chars().next() {
+        let char_len = first_char.len_utf8(); // Get the UTF-8 byte length of the first character
+        Some(&string[..char_len]) // Return the slice that represents the first character
+    } else {
+        None // Return None if the string is empty
+    }
+}
+
+type Student = (String, u32);
+
+/// Given a reference to a student, return the student's name
+fn get_name(classmate: &Student) -> &str {
+    &classmate.0
+}
+
+/// Given a slice of i32s, return the sum of the elements
+fn slice_sum(sliced_ints: &[i32]) -> i32 {
+    sliced_ints.iter().sum()
+}
+
+/// Given a string slice, look for a substring, and return a slice of the first
+/// occurrence of the substring (return None if the substring is not found)
+fn find_in_string<'a>(big_str: &'a str, sub_str: &'a str) -> Option<&'a str> {
```

Comment:
add the `'a` lifetime tag only to the big string (you want the substring to be dropped with the function)